### PR TITLE
VEGA-1453 Retrieve case data from the payment api endpoint

### PR DESCRIPTION
**Pull request by hiboabd**
Case id is now returned along with the payment information when retrieving a payment by id. This is to prevent users from potentially being able to amend the url param to return payment data for a different case and display unrelated case data for the payment.

#minor

### CODE REVIEW

File: internal/server/edit_payment.go [hunk 1]
```diff
@@ -28,7 +28,7 @@ type editPaymentData struct {
 
 func EditPayment(client EditPaymentClient, tmpl template.Template) Handler {
 	return func(w http.ResponseWriter, r *http.Request) error {
-		caseID, err := strconv.Atoi(r.FormValue("id"))
+		_, err := strconv.Atoi(r.FormValue("id"))
```
Comment (hiboabd) [inline internal/server/edit_payment.go:31]: I've left this in to validate the url parameter although the parameter is no longer used

File: internal/sirius/payment.go [hunk 2]
```diff
@@ -10,6 +10,7 @@ type Payment struct {
 	Source      string     `json:"source"`
 	Amount      int        `json:"amount"`
 	PaymentDate DateString `json:"paymentDate"`
+	Case        *Case      `json:"case,omitempty"`
```
Comment (gregtyler) [inline internal/sirius/payment.go:13]: I'm still not very confident about pointers in Go, but we use direct references elsewhere so I'd assume we should do the same here:
```suggestion
	Case        Case       `json:"case,omitempty"`
```

File: internal/server/edit_payment.go (hunk 1 above)
Comment (GemTay) [reply internal/server/edit_payment.go:31]: I dont think this shouldn't be needed anymore, we should only expect the payment ID url param now :)

File: internal/sirius/payment.go (hunk 2 above)
Comment (hiboabd) [reply internal/sirius/payment.go:13]: Thanks i'm not that knowledgable on pointers but `omitempty` doesn't work with structs as they don't have any [empty value](https://www.sohamkamani.com/golang/omitempty/) - and the article uses pointers as a solution to the problem

Comment (gregtyler) [reply internal/sirius/payment.go:13]: Oooo yes, very clever 👍🏽

**Review by GemTay — APPROVED** [review]

**Pull request merged**